Desktop/AddGoalWindow.cs

```diff
@@ -5,23 +5,86 @@
 using System.Drawing;
 using System.Text;
 using System.Windows.Forms;
+using SmartSaver.Data;
 using SmartSaver.Models;
 
 namespace SmartSaver.Desktop
 {
+  
     public partial class AddGoalWindow : Form
     {
         public AddGoalWindow()
         {
             InitializeComponent();
         }
 
+
+        private void goalAmount_KeyPress(object sender, KeyPressEventArgs e)
+        {
+            char ch = e.KeyChar;
+
+            if (ch == 46 && goalMoney.Text.IndexOf('.') != -1)
+            {
+                e.Handled = true;
+                return;
+            }
+            if (!Char.IsDigit(ch) && ch != 8 && ch != 46)
+            {
+                e.Handled = true;
+            }
+        }
```

Comment:
Maybe it could be done in shorter way?

Reply:
mhm.. I don't think there's a point of short it